### PR TITLE
feat: [PIE-4404]: status block new design with icons and new mapping for colors

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

- status block with icons
- new mapping for colors <-> status
- moved mapping from css -> JS

[Designs](https://www.figma.com/file/4HavSweFhZeVsJoaWwSrj8/Pipelines?node-id=0%3A1)

#### Screenshots
**All possible status**
Screenshot 2022-07-12 at 10 28 10 AM

**Hover state**
Screenshot 2022-07-12 at 8 18 13 AM

**code source hover**
Screenshot 2022-07-12 at 8 18 59 AM

**Execution console view for stages**
Screenshot 2022-07-12 at 8 17 52 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/72-triggers/strings/strings.en.yaml [hunk 1]
```diff
@@ -68,7 +68,6 @@ tooltip:
   defaultGitSyncBranchOnly: Triggers can be added to a Pipeline in the default branch only
 triggerConfigurationPanel:
   title: 'Trigger Configuration: On New Webhook'
-  triggerName: 'Trigger Name'
```
Comment (udayvunnam) [inline src/modules/72-triggers/strings/strings.en.yaml:71]: This is moved to commons to use between triggers and pipeline

**Comment by udayvunnam** [comment]
retrigger titlecheck

**Comment by udayvunnam** [comment]
retrigger titlecheck

**Review by vkbansal — DISMISSED** [review]

**Comment by udayvunnam** [comment]
retrigger jest

**Comment by udayvunnam** [comment]
retrigger typecheck

File: src/modules/70-pipeline/components/StatusHeatMap/StatusHeatMap.tsx [hunk 2]
```diff
@@ -1,12 +1,51 @@
...
+ */
+
+import { Color, Icon, IconName, Popover, PopoverProps } from '@harness/uicore'
 import cx from 'classnames'
 import React from 'react'
+import type { ExecutionStatus } from '@pipeline/utils/statusHelpers'
 import css from './StatusHeatMap.module.scss'
 
+const stateMap: Record<string, ExecutionStatus[]> = {
+  success: ['Success', 'IgnoreFailed'],
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/components/StatusHeatMap/StatusHeatMap.tsx:15]: Avoid hardcoding strings here. Use ExecutionStatusEnum present in src/modules/70-pipeline/utils/statusHelpers.ts

File: src/modules/70-pipeline/components/StatusHeatMap/StatusHeatMap.tsx [hunk 3]
```diff
@@ -1,12 +1,51 @@
...
+  running: ['Running', 'AsyncWaiting', 'TaskWaiting', 'TimedWaiting']
+}
+export const getStatusMapping = (status: ExecutionStatus) => {
+  // ['Skipped,Queued,Discontinuing,NotStarted'] or any other unknown status will default to this
+  const colorMap = {
+    primaryState: 'default',
+    icon: '' as IconName | '',
+    iconColor: ''
+  }
+  if (stateMap.success.includes(status)) {
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/components/StatusHeatMap/StatusHeatMap.tsx:27]: Suggestion: We can make the search faster by using objects here. 
const stateMap = {
  success: {
    'Success': 1
  }
}

if (stateMap.success[status]) { // logic }

Comment (depanshu-dhiman) [reply src/modules/70-pipeline/components/StatusHeatMap/StatusHeatMap.tsx:27]: You may also use already defined helper.
Eg. isExecutionSuccess present in src/modules/70-pipeline/utils/statusHelpers.ts

File: src/modules/70-pipeline/pages/pipeline-list/PipelineListTable/PipelineListCells.tsx [hunk 4]
```diff
@@ -252,37 +264,45 @@ export const PipelineNameCell: Renderer<CellProps<PMSPipelineSummaryResponse>> =
...
-    'errored',
-    'suspended',
-    'success',
-    'ignorefailed',
-    'running',
-    'resourcewaiting',
-    'inputwaiting',
-    'paused'
+  const statuses: ExecutionStatus[] = [
+    'Skipped',
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/pages/pipeline-list/PipelineListTable/PipelineListCells.tsx:271]: Please use enum here as well

**Review by depanshu-dhiman — CHANGES_REQUESTED** [review]
Please address the comments.

File: src/modules/70-pipeline/pages/pipeline-list/PipelineListTable/PipelineListCells.tsx (hunk 4 above)
Comment (udayvunnam) [reply src/modules/70-pipeline/pages/pipeline-list/PipelineListTable/PipelineListCells.tsx:277]: This is temporary(commented in code), I am just forming data before it's available from API, the route is not publicly exposed yet.

File: src/modules/70-pipeline/components/StatusHeatMap/StatusHeatMap.tsx (hunk 3 above)
Comment (udayvunnam) [reply src/modules/70-pipeline/components/StatusHeatMap/StatusHeatMap.tsx:46]: I didn't get the first comment on how we could use that logic for all 4 varying conditions

```
if (stateMap.success.includes(status)) {
    colorMap.primaryState = 'success'
  } else if (stateMap.failed.includes(status)) {
    colorMap.primaryState = 'failed'
    colorMap.icon = 'cross'
    colorMap.iconColor = Color.RED_500
  } else if (stateMap.paused.includes(status)) {
    colorMap.primaryState = 'paused'
    colorMap.icon = 'pause'
    colorMap.iconColor = Color.ORANGE_500
  } else if (stateMap.running.includes(status)) {
```

**Review by depanshu-dhiman — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**